client/wc-api/items/operations.js

```diff
@@ -0,0 +1,65 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * WooCommerce dependencies
+ */
+import { stringifyQuery } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import { getResourceIdentifier, getResourcePrefix, getResourceName } from '../utils';
+import { NAMESPACE } from '../constants';
+
+const typeEndpointMap = {
+	'items-query-categories': 'products/categories',
+	'items-query-customers': 'customers',
+	'items-query-coupons': 'coupons',
+	'items-query-products': 'products',
+	'items-query-taxes': 'taxes',
+};
+
+function read( resourceNames, fetch = apiFetch ) {
+	const filteredNames = resourceNames.filter( name => {
+		const prefix = getResourcePrefix( name );
+		return Boolean( typeEndpointMap[ prefix ] );
+	} );
+
+	return filteredNames.map( async resourceName => {
+		const prefix = getResourcePrefix( resourceName );
+		const endpoint = typeEndpointMap[ prefix ];
+		const query = getResourceIdentifier( resourceName );
+		const url = NAMESPACE + `/${ endpoint }${ stringifyQuery( query ) }`;
+
+		try {
+			const items = await fetch( {
+				path: url,
+			} );
```

Comment:
Kind of the edge case we were talking about before where a query might return more than one page of results, should we somehow expose `X-WP-Total` or `X-WP-TotalPages` so we could display messaging that more results exist?

Reply:
Totally fine with this being follow-up btw, just thinking out loud.

Reply:
Right, actually I think it shouldn't be that difficult making several consecutive queries when there is more than one page of results, similarly as what we do for the chart.

I think that would be a better solution than adding a message in the UI, what do you think?

For now, I added a note at the top comment of this PR and will create a new issue for it when we merge this one.

Reply:
I still think we just have a hard limit of the number of results we fetch and be okay with it, but perhaps add some messaging that "x more results match"